sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -77,6 +77,7 @@ case class DecimalType(precision: Int, scale: Int) extends FractionalType {
       (precision - scale) >= (dt.precision - dt.scale) && scale >= dt.scale
     case dt: IntegralType =>
       isWiderThan(DecimalType.forType(dt))
+    // For DoubleType/FloatType, the value can be NaN, PositiveInfinity or NegativeInfinity.
```

Comment:
This PR also added comment to explain why we can't upcast double/float to decimal.